src/tensors/diagonal.jl

```diff
@@ -80,6 +94,14 @@ end
 function Base.convert(D::Type{<:DiagonalTensorMap}, d::DiagonalTensorMap)
     return DiagonalTensorMap(convert(storagetype(D), d.data), d.domain)
 end
+function Base.convert(::Type{DiagonalTensorMap}, t::AbstractTensorMap)
+    all(LinearAlgebra.isdiag ∘ last, blocks(t)) ||
```

Comment:
```suggestion
    LinearAlgebra.isdiag(t) ||
```
This function is actually defined, I implemented these a while ago: https://github.com/Jutho/TensorKit.jl/blob/450ff9bf21176859123ecfa39b3e769e571a4a9d/src/tensors/linalg.jl#L168-L176
They might require some revisions, since this predates `DiagonalTensorMap` (would also fix some todos)